Azure CLI authentication using Service Principal does not add expiration information, causing NumberFormatException

**Describe the bug**
It is a very common use case that the environment running your K8s automation is a server or system that requires programmatic access. For Azure and Azure Kubernetes Service one of these scenarios is when using a Service Principal for authentication. A Service Principal is basically an AD user that represents an App Registration in Azure. Permissions can be assigned to the service principal in IAM for the AKS cluster.

We utilize Service Principal and the preferred method of authentication is certificate based. See: https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli#sign-in-with-a-service-principal

This is how Azure recommends configuring/authenticating your kubeconfig for your AKS cluster after you have logged in to the azure cli: https://docs.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az-aks-get-credentials

Once you have performed `az aks get-credentials` using the authenticated service principal, then the resulting `~/.kube/config` is missing the following values:
* `expires-in`
* `expires-on`

This is causing our application to receive the following error:
```
2022-08-02 14:42:24,371,/10.80.2.170,,,[XNIO-1 task-1],KubernetesUtil,Switching to context aks-cluster-eastus2
2022-08-02 14:42:24,395,/10.80.2.170,,,[XNIO-1 task-1],ApplicationManagerFilter,Unknown Error

Caused by: java.lang.NumberFormatException: null
        at java.base/java.lang.Long.parseLong(Long.java:655)
        at java.base/java.lang.Long.parseLong(Long.java:817)
        at io.kubernetes.client.util.authenticators.AzureActiveDirectoryAuthenticator.isExpired(AzureActiveDirectoryAuthenticator.java:53)
        at io.kubernetes.client.util.KubeConfig.getCredentials(KubeConfig.java:219)
        at io.kubernetes.client.util.credentials.KubeconfigAuthentication.<init>(KubeconfigAuthentication.java:59)
        at io.kubernetes.client.util.ClientBuilder.kubeconfig(ClientBuilder.java:299)
        at com.application.manager.resource.KubernetesUtil.switchToContext(KubernetesUtil.java:42)
        at com.application.manager.resource.EnterpriseResource.updateApplication(EnterpriseResource.java:614)
```

We believe that not all credentials must have an expiration configured in the kube config file. I cannot find much in the kube config spec / documentation stating that's a requirement.

I am not certain if it is the client's intention to be enforcing the need for an expiration in the config, but it currently does that.
* https://github.com/kubernetes-client/java/blob/b570730dbd50a1c809712f1792302cbc9c9f117e/util/src/main/java/io/kubernetes/client/util/KubeConfig.java#L219
* https://github.com/kubernetes-client/java/blob/f20788272291c0e79a8c831d8d5a7dd94d96d2de/util/src/main/java/io/kubernetes/client/util/authenticators/AzureActiveDirectoryAuthenticator.java#L51

There is no null condition handling here for the `config.get(EXPIRES_ON)`.

**Client Version**
16.0.0

**Kubernetes Version**
1.23.8

**Java Version**
adoptopenjdk/openjdk11:ubuntu-slim

**To Reproduce**
Steps to reproduce the behavior:
```
az login --service-principal -u <app-id> -p <password-or-cert> --tenant <tenant>
az aks get-credentials --name MyManagedCluster --resource-group MyResourceGroup
```
Then attempt to authenticate the Kubernetes Client to your cluster.

**Expected behavior**
I would expect there to be a null condition check and either throw a proper exception if the client intends to enforce the requirement of expiration in the config, or do a null check and only check the expiration if it exists.

**KubeConfig**
I don't think this is needed. Just remove the "expires-on" config from your kubeconfig.

**Server (please complete the following information):**
 - OS: Ubuntu (slim)
 - Environment: Docker container running on Aws EC2 server, connecting to Azure Kubernetes Service API
 - Cloud: Management environment is in AWS EC2, target cluster for management is in Azure AKS

--- a/util/src/main/java/io/kubernetes/client/util/authenticators/AzureActiveDirectoryAuthenticator.java
+++ b/util/src/main/java/io/kubernetes/client/util/authenticators/AzureActiveDirectoryAuthenticator.java
@@ -50,6 +50,9 @@ public class AzureActiveDirectoryAuthenticator implements Authenticator {
   @Override
   public boolean isExpired(Map<String, Object> config) {
     String expiresOn = (String) config.get(EXPIRES_ON);
+    if (expiresOn == null || expiresOn.length() == 0) {
+      return true;
+    }
     Date expiry = new Date(Long.parseLong(expiresOn) * 1000);
     return expiry.compareTo(new Date()) <= 0;
   }

